Scripts/Runtime/Entities/TaskSystem/TaskData/DataStream/Abstract/AbstractArrayDataStream.cs

```diff
@@ -0,0 +1,142 @@
+using Anvil.Unity.DOTS.Data;
+using Anvil.Unity.DOTS.Jobs;
+using Unity.Burst;
+using Unity.Collections;
+using Unity.Jobs;
+#if DEBUG
+using Unity.Profiling;
+#endif
+
+#if ANVIL_DEBUG_LOGGING_EXPENSIVE
+using UnityEngine;
+#endif
+
+namespace Anvil.Unity.DOTS.Entities.Tasks
+{
+    internal abstract class AbstractArrayDataStream<T> : AbstractTypedDataStream<T>,
+                                                         IArrayDataStream
+        where T : unmanaged
+    {
+        //Deliberately not using pointers because that messes up what the safety handle pointer points to.
+        protected DeferredNativeArray<T> Live;
```

Comment:
Not a fan of this name. `Current`, `Active`, `Data`?

Should also be `m_` prefixed right?

Reply:
I'll go with `Active`. 

With the `m_` prefix, yes it should technically. I wanted it to be a getter but that messes up the safety pointer. So this was a bit of a special case. I'm not sure it's still relevant but I'll save this to see what I can do.

Reply:
May be able to do a `ref` getter to help with the safety pointer.